src/legacy/core_plugins/markdown_vis_type/public/index.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { markdownVis } from './markdown_vis';
+// @ts-ignore
+import { kibanaMarkdown } from './markdown_fn';
+
+// @ts-ignore
+import { functionsRegistry } from '../../interpreter/public/registries';
+import { visualizations, VisualizationsSetup } from '../../visualizations/public';
+
+class MarkdownVisTypePlugin {
+  constructor() {}
+
+  public setup(visualizationsSetup: VisualizationsSetup) {
+    visualizationsSetup.types.VisTypesRegistryProvider.register(() => markdownVis);
+    functionsRegistry.register(kibanaMarkdown);
+  }
+
```

Comment:
It looks like core's `plugin_service` calls all `setup()`, `start()` and `stop()` methods without checking if they exist, thus we need to provide `start()` here, too.

This is also one reason why it is useful to create real NP plugins; as only then we can really know if they work.